cdap-ui/app/features/adapters/controllers/create-new-ctrl.js

```diff
@@ -78,7 +78,33 @@ angular.module(PKG.name + '.feature.adapters')
         }.bind(this));
     };
 
+
+    var confirmOnPageExit = function (e) {
+
+      if (!MyPlumbService.isConfigTouched) { return; }
+      // If we haven't been passed the event get the window.event
+      e = e || window.event;
```

Comment:
May be we should use ```$window```? If at all when we add tests it will be easier to test. I think we have been using ```$window``` in other places I hope.